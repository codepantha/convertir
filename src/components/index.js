export { default as Card } from './card/Card';
export { default as Header } from './header/Header';
export { default as Navigation } from './navigation/Navigation';
export { default as Filter } from './filter/Filter';
export { default as Loader } from './loader/Loader';
