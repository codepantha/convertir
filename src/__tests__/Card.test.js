import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Card } from '../components';

const currency = {
  currency: 'usd',
  rate: 20,
};

describe('render Card to screen', () => {
  test('render Card Component', async () => {
    render(<Card currency={currency} />);

    const cardElement = await screen.getByTestId('card');
    expect(cardElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Card currency={currency} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
