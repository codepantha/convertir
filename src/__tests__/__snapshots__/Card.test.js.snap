// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Card to screen matches snapshot 1`] = `
<div
  className="convertir__currency-card shadow-5 rotate-vert-center"
  data-testid="card"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12.0519 14.8285L13.4661 16.2427L17.7088 12L13.4661 7.7574L12.0519 9.17161L13.8804 11H6.34321V13H13.8803L12.0519 14.8285Z"
      fill="currentColor"
    />
    <path
      clipRule="evenodd"
      d="M19.7782 19.7782C24.0739 15.4824 24.0739 8.51759 19.7782 4.22183C15.4824 -0.0739417 8.51759 -0.0739417 4.22183 4.22183C-0.0739417 8.51759 -0.0739417 15.4824 4.22183 19.7782C8.51759 24.0739 15.4824 24.0739 19.7782 19.7782ZM18.364 18.364C21.8787 14.8492 21.8787 9.15076 18.364 5.63604C14.8492 2.12132 9.15076 2.12132 5.63604 5.63604C2.12132 9.15076 2.12132 14.8492 5.63604 18.364C9.15076 21.8787 14.8492 21.8787 18.364 18.364Z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
  <div
    className="convertir__currency-card_info grow"
  >
    <h2>
      usd
    </h2>
    <p>
      20
    </p>
  </div>
</div>
`;
